### PR TITLE
Implementing new strategy token valuation model + balancer oracle removal

### PR DESCRIPTION
**Pull request by weitianjie2000**

### CODE REVIEW

**Comment by T-Woodward** [comment]
I'm not understanding the _validateSpotPriceAndPairPrice function. When you call _checkPriceLimit twice, aren't you duplicating work? Specifically, doesn't this code intend to perform the same function as _getSpotPrice? https://github.com/notional-finance/leveraged-vaults/blob/4b3607529b47f95c7e95238a33e3d14f06ac866f/contracts/vaults/balancer/internal/math/Stable2TokenOracleMath.sol#L101-L117

File: contracts/vaults/balancer/internal/pool/TwoTokenPoolUtils.sol [hunk 1]
```diff
@@ -121,37 +94,29 @@ library TwoTokenPoolUtils {
...
-            // convert secondaryTokenValue to underlyingValue using the pairPrice.
-            // Both bptPrice and bptAmount are in 1e18
-            uint256 secondaryAmount = (bptPrice * bptAmount) / BalancerConstants.BALANCER_PRECISION;
-
-            // And then normalizing to primary token precision we add:
-            // PrimaryAmount = (SecondaryAmount * primaryPrecision) / PairPrice
-            primaryAmount = (secondaryAmount * primaryPrecision) / pairPrice;
-        }
+        // Value the secondary balance in terms of the primary token using the oraclePairPrice
+        uint256 secondaryAmountInPrimary = secondaryBalance * BalancerConstants.BALANCER_PRECISION / spotPrice;
```
Comment (T-Woodward) [inline contracts/vaults/balancer/internal/pool/TwoTokenPoolUtils.sol:115]: @weitianjie2000 please use the oraclePairPrice here, not the spotPrice

Comment (weitianjie2000) [reply contracts/vaults/balancer/internal/pool/TwoTokenPoolUtils.sol]: fixed

File: contracts/vaults/balancer/BalancerVaultTypes.sol [hunk 2]
```diff
@@ -184,7 +179,6 @@ struct StrategyVaultSettings {
     /// @notice Slippage limit for selling reward tokens
     uint32 maxRewardTradeSlippageLimitPercent;
     uint16 maxBalancerPoolShare;
-    uint16 balancerOracleWeight;
```
[the remark this answers is not in the document]
Comment (weitianjie2000) [reply contracts/vaults/balancer/BalancerVaultTypes.sol:186]: will deploy a fresh vault

**Pull request merged**